### PR TITLE
389 dashboard matrix queries

### PR DESCRIPTION
**Pull request by JackVCurtis**
Tickets: 
- https://github.com/orgs/AMP-SCZ/projects/2/views/3?pane=issue&itemId=37850197
- https://github.com/orgs/AMP-SCZ/projects/2/views/3?pane=issue&itemId=37850716
- https://github.com/orgs/AMP-SCZ/projects/2/views/3?pane=issue&itemId=37850525

This turned out to be way easier than I thought, which I guess is better than the reverse. Dropping the connection to `dpdata` but leaving the `dataDb` variable in place is ugly but worked great, and the additional migrations were a straight copy from one db to the other. `DashboardDataProcessor` looks complicated, but once I worked out that most of it was happening in memory updating the query was fairly simple. Still lots of work that can be done around end-to-end testing and performance improvements, but this should get us to the first step of having a sensible data structure in Mongo.

### CODE REVIEW

File: server/data_processors/DashboardDataProcessor/index.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ class DashboardDataProcessor {
   #dataFromAssessments = async () =>
     await Promise.all(
       this.#assessmentsFromConfig.map(async ({ collection, assessment }) => {
```
Comment (Elkrival) [inline server/data_processors/DashboardDataProcessor/index.js]:
```suggestion
      this.#assessmentsFromConfig.map(async ({ assessment }) => {
```

Comment (Elkrival) [inline server/data_processors/DashboardDataProcessor/index.js]: I think now we don't need the #assessmentsFromConfig we can use the config directly and we don't need to append the collection property anymore. so the data result is all we would need now for the query, perhaps we should be careful about confits with the same assessment name..? Perhaps a new list of just assessments from the config for a query that looks like. 

```javascript
const assessmentsFromConfig = config.map(({analysis} => analysis)

this.db.collection(collections.assessmentSubjectDayData)
          .find({ subject: req.params.subject, assessment: { $in: assessmentsFromConfigs }).toArray()
```
```suggestion
  #dataFromAssessments = async () =>
    await Promise.all(
      this.#assessmentsFromConfig.map(async ({ collection, assessment }) => {
```

File: scripts/data_migration/migrate_metadata.test.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import { ObjectId } from 'mongodb';
+
+import { migrateMetadata } from './migrate_metadata';
+
+describe('migrateMetadata', () => {
```
Comment (Elkrival) [inline scripts/data_migration/migrate_metadata.test.js:5]: 👍

File: server/data_processors/DashboardDataProcessor/DashboardDataProcessor.test.js [hunk 3]
```diff
@@ -49,43 +50,41 @@ describe('Data Processors - Dashboard', () => {
         category: 'reading',
       }),
     ]
-    const db = createDb()
     const jumpVariableData = [
       createSubjectDayData({
+        collection: 'jump-of-collection',
```
Comment (Elkrival) [inline server/data_processors/DashboardDataProcessor/DashboardDataProcessor.test.js]: this property shouldn't exist in the subjects data

**Review by Elkrival — COMMENTED** [review]
A slight refactor of some code that we will no longer need.

File: server/data_processors/DashboardDataProcessor/index.js (hunk 1 above)
Comment (JackVCurtis) [reply server/data_processors/DashboardDataProcessor/index.js]: I think you're right but there's a lot that can be done to improve performance and legibility here that I don't want to block being able to update the underlying collection. My preference would be to handle this as part of a future performance improvement PR.

Comment (Elkrival) [reply server/data_processors/DashboardDataProcessor/index.js]: that makes sense, but I do feel like we aren't missing the subject in the query

Comment (JackVCurtis) [reply server/data_processors/DashboardDataProcessor/index.js]: e42ee3084c5d27c1c4ba2f3b3590560d8f5fa3bf

File: server/data_processors/DashboardDataProcessor/DashboardDataProcessor.test.js (hunk 3 above)
Comment (JackVCurtis) [reply server/data_processors/DashboardDataProcessor/DashboardDataProcessor.test.js]: e42ee3084c5d27c1c4ba2f3b3590560d8f5fa3bf

**Review by Elkrival — APPROVED** [review]
👍

**Pull request merged**